cmd/thv-operator/controllers/mcpremoteproxy_deployment.go

```diff
@@ -80,10 +80,44 @@ func (r *MCPRemoteProxyReconciler) deploymentForMCPRemoteProxy(
 }
 
 // buildContainerArgs builds the container arguments for the proxy
-func (*MCPRemoteProxyReconciler) buildContainerArgs() []string {
+func (*MCPRemoteProxyReconciler) buildContainerArgs(ctx context.Context, proxy *mcpv1alpha1.MCPRemoteProxy) []string {
 	// The third argument is required by proxyrunner command signature but is ignored
 	// when RemoteURL is set (HTTPTransport.Setup returns early for remote servers)
-	return []string{"run", "--foreground=true", "placeholder-for-remote-proxy"}
+	args := []string{"run", "--foreground=true", "placeholder-for-remote-proxy"}
+
+	// Add user-specified proxy args from ResourceOverrides
+	// These are inserted after "run" but before other arguments
+	if proxy.Spec.ResourceOverrides != nil &&
+		proxy.Spec.ResourceOverrides.ProxyDeployment != nil &&
+		len(proxy.Spec.ResourceOverrides.ProxyDeployment.Args) > 0 {
+
+		// Insert additional args between "run" and "--foreground=true"
+		// Current structure: ["run", "--foreground=true", "placeholder-for-remote-proxy"]
+		// We want: ["run", "--debug", "--foreground=true", "placeholder-for-remote-proxy"]
+		// So we insert the override args after "run" (position 1)
+		insertPosition := 1
+
+		// Prevent size computation overflow by ensuring the total number of args is within safe bounds.
+		const maxAllowedArgs = 10000
```

Comment:
Do the ordering of flags matter? Also not sure we have to protect against 10000 flags?